api/channel.go

```diff
@@ -397,10 +397,15 @@ func updateChannelPurpose(c *Context, w http.ResponseWriter, r *http.Request) {
 }
 
 func getChannels(c *Context, w http.ResponseWriter, r *http.Request) {
+	// check if the team exist
+	if result := <-Srv.Store.Team().Get(c.TeamId); result.Err != nil {
```

Comment:
Actually, I'm changing my mind (I think),  probably makes sense to put it into context.go.  Not sure if you even need todo the check for system admin and load the team since it should be in `c.Session.GetTeamByTeamId` If you're already a member of the team.